packages/core/src/browser/mime-service.ts

```diff
@@ -0,0 +1,79 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, injectable, interfaces } from 'inversify';
+import {
+    createPreferenceProxy,
+    FrontendApplication,
+    FrontendApplicationContribution,
+    PreferenceChange,
+    PreferenceContribution,
+    PreferenceProxy,
+    PreferenceSchema,
+    PreferenceService
+} from '.';
+import { MaybePromise } from '../common/types';
+
+@injectable()
+export class MimeService implements FrontendApplicationContribution {
+    public static readonly FILES_ASSOCIATIONS_PREFERENCE: string = 'files.associations';
+
+    @inject(PreferenceService)
+    protected readonly preferenceService: PreferenceService;
+
+    onStart?(app: FrontendApplication): MaybePromise<void> {
+        this.preferenceService.onPreferenceChanged((e: PreferenceChange) => {
+            if (e.affects(MimeService.FILES_ASSOCIATIONS_PREFERENCE)) {
+                this.updateMime();
+            }
+        });
+    }
+
+    protected updateMime() {
+        // stub
+    }
+}
+
+export const filesystemAssociationsPreferenceSchema: PreferenceSchema = {
```

Comment:
it should go into `@theia/filesystem` extension, into `FileSystemPreference`

`core` should not know anything about filesystem support, only provide a stub for `MimeService`